bfdd/bfd_packet.c

```diff
@@ -630,10 +630,10 @@ int bfd_recv_cb(struct thread *t)
 	 * Single hop: set local address that received the packet.
 	 */
 	if (is_mhop) {
-		if ((BFD_TTL_VAL - bfd->mh_ttl) > BFD_TTL_VAL) {
+		if (ttl < bfd->mh_ttl) {
 			cp_debug(is_mhop, &peer, &local, ifindex, vrfid,
 				 "exceeded max hop count (expected %d, got %d)",
-				 bfd->mh_ttl, BFD_TTL_VAL);
+				 bfd->mh_ttl, ttl);
```

Comment:
Wouldn't it be better to use the `IP_MINTTL`/`IPV6_MINHOPCOUNT` sockoptions to let the kernel do this work for us? We already have the `sockopt_minttl()` wrapper to set them on any socket.

IIRC GTSM for LDP requires this sort of filtering to be done on the forwarding plane to protect the control plane from DoS attacks. Not sure what the BFD specs say though.

Reply:
There is a possibility for that, but I guess it wouldn't work: the multi hop socket that receives packet is the same for all connections and I guess we can't bind multiple sockets to `INADDR_ANY` and then we'd need one socket per TTL value (or open them dynamically). Another option is to bind one socket per local address, but then it would complicate the code for peers on the same interface.

Remembering: there is only one port for destination UDP for BFD and it is multiplexed for all sessions.

Reply:
yeah in that case using those sockoptions would be impractical since we have the same socket shared by multiple peers. Marking as resolved...